dashboard/cron.py

```diff
@@ -226,24 +230,25 @@ def update_with_bq_access(self):
                     for table in tables:
```

Comment:
in the function, `update_with_bq_access` from line 220 - 272 need to be chunked to the small set of pieces. It's almost unreadable unless you pay close attention.

Reply:
Yeah, it is complex, but I didn't write it the first time! ;) Some of these if's are probably unnecessary. 

I think this will have to be rewritten completely to be able to handle other BQ calls like for textbooks, just wanted to make this change now (to retrieve records in bulk) as it's a cost/time savings. I think we could address the method complexity in #402

Reply:
I also feel like we might be "close" to having unittests for some of these methods too, when the app has more mock data. That would make large refactors a little safer.

Reply:
I agree, unit test make it easier to test and we don't need to rely on functional testing each time

Reply:
I rewrote this whole method eliminating most of the if statements as they looked to be unnecessary and were actually outdated based on the recent changes. I'm not sure why there were in there in the first place. Maybe it looks better for you now?